### PR TITLE
[flink] Support setting upper limit for inferring scan parallelism for source

### LINKED ISSUE #2205: [Feature] Support setting upper limit for inferring scan parallelism for source
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Currently, Paimon support infer scan parallelism from splits number or bucket number https://github.com/apache/incubator-paimon/pull/67, it's better to limit the parallelism to prevent source parallelism of flink job is too large.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by damumu0625**
### Purpose

Linked issue: close #2205

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java [hunk 1]
```diff
@@ -120,6 +120,13 @@ public class FlinkConnectorOptions {
                             "If it is false, parallelism of source are set by global parallelism."
                                     + " Otherwise, source parallelism is inferred from splits number (batch mode) or bucket number(streaming mode).");
 
+    public static final ConfigOption<Integer> INFER_SCAN_MAX_PARALLELISM =
+            ConfigOptions.key("scan.infer-parallelism.max")
+                    .intType()
+                    .defaultValue(Integer.MAX_VALUE)
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java:126]: default 1000?

**Review by JingsongLi — COMMENTED** [review]
Thanks @damumu0625 
You need to generate documentation too, see paimon-docs/README.md

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java (hunk 1 above)
Comment (damumu0625) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java:126]: The default value is set to 1024

**Comment by JingsongLi** [comment]
@damumu0625 Please modify `FlinkConnectorOptions` default value too.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**